tests/plugins/colordialog/_helpers/tools.js

```diff
@@ -0,0 +1,40 @@
+/* exported assertColor, assertColorAtDialogShow */
+
+function assertColor( editor, inputColor, outputColor ) {
+	editor.once( 'dialogShow', function( evt ) {
+		var dialog = evt.data;
+		dialog.setValueOf( 'picker', 'selectedColor', inputColor );
+		dialog.getButton( 'ok' ).click();
+	} );
+
+	editor.getColorFromDialog( function( color ) {
+		resume( function() {
+			assert.areSame( outputColor, color );
+		} );
+	} );
+	wait();
+}
+
+function assertColorAtDialogShow( editor, expectedColor, html, button ) {
+	var toolbarButton = editor.ui.get( button );
+	editor.insertHtml( html );
+	editor.once( 'dialogShow', function( evt ) {
+		resume( function() {
+			var dialog = evt.data,
+				selectedColor = dialog.getValueOf( 'picker', 'selectedColor' );
+			dialog.getButton( 'ok' ).click();
+
+			assert.areSame( expectedColor, selectedColor );
+		} );
+	} );
+
+	toolbarButton.click( editor );
+	openColorDialog( toolbarButton );
+}
+
+function openColorDialog( button ) {
```

Comment:
Can't `bot.dialog` be used for it?

Reply:
Same as above, using `bot.dialog()` doesn't trigger some logic responsible for reading the color from selection. I even wrote such tests [here](https://github.com/ckeditor/ckeditor-dev/pull/2934/commits/b8e67ad706269cf1631138959cc573874d1b3f81) but later realised that I have to pass color by myself so it's basically simplified and poor version of previous tests and deleted them as redundant [here](https://github.com/ckeditor/ckeditor-dev/pull/2934/commits/77a929f211f5e50d2c34516273a5489806b9b122).